acceptance/suite_test.go

```diff
@@ -0,0 +1,261 @@
+package influxtest
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ahmetalpbalkan/dlog"
+	"github.com/colega/envconfig"
+	influxdb "github.com/influxdata/influxdb-client-go/v2"
+	influxdb_api "github.com/influxdata/influxdb-client-go/v2/api"
+	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
+	promapi "github.com/prometheus/client_golang/api"
+	promv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/stretchr/testify/suite"
+)
+
+var (
+	suiteContainerLabels = map[string]string{"acceptance-suite": "influx-proxy"}
+)
+
+func TestSuite(t *testing.T) {
+	suite.Run(t, new(Suite))
+}
+
+type Suite struct {
+	suite.Suite
+
+	// cfg is filled with env values starting with ACCEPTANCE_ prefix,
+	// for example, ACCEPTANCE_DOCKER_HOST for cfg.Docker.Host
+	cfg struct {
+		// CI should be set to true in CI env
+		CI     bool
+		Docker struct {
+			// Host defines where docker containers run
+			// This is `localhost` for local env, but in CI containers run in a dind service that runs on `docker` host.
+			Host string `default:"localhost"`
+			// Tag defines which version of our apps has to be tested. In local, these are built using `make build-local`,
+			// while in CI those are images pushed to gcr ready to be used in production.
+			Tag string `default:"local"`
+			// Auth is used in CI to be able to pull images from GCR. In local we expect the images to be already built.
+			Auth docker.AuthConfiguration
+		}
+	}
+
+	pool    *dockertest.Pool
+	network *dockertest.Network
+
+	cortexResource      *dockertest.Resource
+	influxProxyResource *dockertest.Resource
+
+	api struct {
+		influx_client influxdb.Client
+		writeAPI      influxdb_api.WriteAPIBlocking
+		promClient    promapi.Client
+		querierClient promv1.API
+	}
+
+	suiteReady time.Time
+}
+
+func (s *Suite) SetupSuite() {
```

Comment:
can you add a docstring to this function that describes what services are started as part of the acceptance tests?  Also probably add a top-level docstring describing the test approach as a whole